cmd/registry/cmd/compute-references.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/apigee/registry/cmd/registry/core"
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/rpc"
+	"github.com/apigee/registry/server/names"
+	openapi_v2 "github.com/googleapis/gnostic/openapiv2"
+	openapi_v3 "github.com/googleapis/gnostic/openapiv3"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/proto"
+)
+
+func init() {
+	computeCmd.AddCommand(computeReferencesCmd)
+}
+
+var computeReferencesCmd = &cobra.Command{
+	Use:   "references",
+	Short: "Compute references of API specs",
+	Args:  cobra.MinimumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.TODO()
+		client, err := connection.NewClient(ctx)
+		if err != nil {
+			log.Fatalf("%s", err.Error())
+		}
+		// Initialize task queue.
+		taskQueue := make(chan core.Task, 1024)
+		workerCount := 64
+		for i := 0; i < workerCount; i++ {
+			core.WaitGroup().Add(1)
+			go core.Worker(ctx, taskQueue)
+		}
+		// Generate tasks.
+		name := args[0]
+		if m := names.SpecRegexp().FindStringSubmatch(name); m != nil {
+			// Iterate through a collection of specs and compute references for each
+			err = core.ListSpecs(ctx, client, m, computeFilter, func(spec *rpc.ApiSpec) {
+				taskQueue <- &computeReferencesTask{
+					ctx:      ctx,
+					client:   client,
+					specName: spec.Name,
+				}
+			})
+			if err != nil {
+				log.Fatalf("%s", err.Error())
+			}
+			close(taskQueue)
+			core.WaitGroup().Wait()
+		}
+	},
+}
+
+type computeReferencesTask struct {
+	ctx      context.Context
+	client   connection.Client
+	specName string
+}
+
+func (task *computeReferencesTask) Name() string {
```

Comment:
This name might be an established pattern already, but FYI this function seems like it might be useful as `String() string` where it's recognized and used by formatters and other utilities for printing the string version of a task.

Reply:
I agree. Let's change them all in a single (future) PR.